### DISCUSSION TITLE
Custom font/ CSS in Email Template for authentication

### DISCUSSION THREAD

**Discussion by Criepstar** [supabase/supabase#2831] [Questions] [answered]
Hello would it be possible to have custom fonts via CSS in the Email templates of Supabase?

**Comment by inian** [comment, marked as answer]
Hi @Criepstar, we will bring in a more powerful template editor at a later date, but for now you can get the confirmation URLs from the gotrue server directly. With this, you can use a third party tool to generate HTML emails and embed the confirmation URLs in the templates programatically - [docs](https://supabase.io/blog/2021/07/28/supabase-auth-passwordless-sms-login#generate-confirmation-links)

**Comment by gokulkrishh** [comment]
@inian Any update on this?

**Reply by inian** [reply]
We dont have a template editor yet Gokul, the confirmation URL API mentioned above is the workaround for bringing your own templates

**Reply by gokulkrishh** [reply]
@inian Okay. I will check the above workaround. Thanks. B/w I absolutely love supabase. Keep doing the awesome work.

**Comment by rodrigomf24** [comment]
@inian I am using mjml to generate the email html, but when I insert it into the email template text area in supabase it strips out the style tags, link tags, and simply leaves the html inside the body tag. Is there a way to persist or retain the header tag content?

**Comment by garylake** [comment]
Just bumping this one as I've stumbled across the same issue as @rodrigomf24 

For context, I'm using postmark and sending my magiclink emails via SMTP. I've got other bespoke transactional emails I'm sending via the Postmark API using /email/withTemplate/. I was hoping I could just paste my postmark templates back into Supabase as a quick solution to getting my Magiclink emails looking consistent with my postmark templates. But Supabase strips the body/head and thus style tag on save. 

Is there a technical reason for this and can we not have an option to let us paste the full HTML template in?

**Reply by cliftonlabrum** [reply]
Ugh... I just ran into this same thing after spending two hours creating an email template. I pasted it into Supabase and it stripped out everything that was in the `<head>` of the template. 😔

**Reply by cliftonlabrum** [reply]
As a workaround, I found this tool from MailChimp that converts all your `<head>` styles into inline styles. It works with Supabase's constraints: https://templates.mailchimp.com/resources/inline-css/

**Reply by bombillazo** [reply]
@cliftonlabrum thanks so much!!!

**Comment by organicnz** [comment]
Hey, guys, you can use some [Figma templates](https://www.figma.com/file/wfBgrMaG0pROmof81rsMM6/Mail-design-system-for-Marka-Plugin-(Community)?node-id=553%3A1431&t=pLpC2xL3zBvDAWlh-1 ) and export everything with Marka Email Generator in a single HTML snippet. I recently found this awesome project and already created several templates for my Supabase app. Also, I've tested it out with Mailgun today hehe :))

<img width="1440" alt="Screenshot 2023-02-24 at 11 11 35 PM" src="https://user-images.githubusercontent.com/4765090/221281668-be74bddc-166f-4afa-8359-6234b3df4a92.png">
